haystack/components/rankers/sentence_transformers_similarity.py

```diff
@@ -0,0 +1,280 @@
+
+    @component.output_types(documents=List[Document])
+    def run(
+        self,
+        *,
+        query: str,
+        documents: List[Document],
+        top_k: Optional[int] = None,
+        scale_score: Optional[bool] = None,
+        score_threshold: Optional[float] = None,
+    ):
+        """
+        Returns a list of documents ranked by their similarity to the given query.
+
+        :param query:
+            The input query to compare the documents to.
+        :param documents:
+            A list of documents to be ranked.
+        :param top_k:
+            The maximum number of documents to return.
+        :param scale_score:
+            If `True`, scales the raw logit predictions using a Sigmoid activation function.
+            If `False`, disables scaling of the raw logit predictions.
+            If set, overrides the value set at initialization.
+        :param score_threshold:
+            Use it to return documents only with a score above this threshold.
+            If set, overrides the value set at initialization.
+        :returns:
+            A dictionary with the following keys:
+            - `documents`: A list of documents closest to the query, sorted from most similar to least similar.
+
+        :raises ValueError:
+            If `top_k` is not > 0.
+        :raises RuntimeError:
+            If the model is not loaded because `warm_up()` was not called before.
+        """
+        if self.model is None:
+            raise RuntimeError(
+                "The component SentenceTransformersSimilarityRanker wasn't warmed up. "
+                "Run 'warm_up()' before calling 'run()'."
+            )
+
+        if not documents:
+            return {"documents": []}
+
+        top_k = top_k or self.top_k
+        scale_score = scale_score or self.scale_score
+        score_threshold = score_threshold or self.score_threshold
+
+        if top_k <= 0:
+            raise ValueError(f"top_k must be > 0, but got {top_k}")
+
+        prepared_query = self.query_prefix + query
+        prepared_documents = []
+        for doc in documents:
+            meta_values_to_embed = [
+                str(doc.meta[key]) for key in self.meta_fields_to_embed if key in doc.meta and doc.meta[key]
+            ]
+            prepared_documents.append(
+                self.document_prefix + self.embedding_separator.join(meta_values_to_embed + [doc.content or ""])
+            )
+
+        activation_fn = Sigmoid() if scale_score else Identity()
+
+        ranking_result = self.model.rank(
+            query=prepared_query,
+            documents=prepared_documents,
+            batch_size=self.batch_size,
+            activation_fn=activation_fn,
+            convert_to_numpy=True,
+            return_documents=False,
+        )
+
+        ranked_docs = []
+        for el in ranking_result:
+            index = el["corpus_id"]
+            score = el["score"]
+            document = copy(documents[index])
```

Comment:
I thought we didn't need to copy because at the Pipeline level we deepcopy all inputs before passing them to the component?

Reply:
In general, I would prefer to copy documents.
If the component is used outside of a Pipeline, it would modify documents in place, which is unexpected.
`copy` should be lightweight and I have verified that it works properly on `Document` objects.
OK for you?

Reply:
Sounds good, probably worth noting that down internally if that's a general policy we'd like to follow.